### PR TITLE
Add missing time update for uf

### PR DESCRIPTION
**Pull request by ArnoStrouwen**
If this is a bug, it is pretty bad it slips past tests.

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Yeah... could you add a test that catches this?

File: test/forward.jl [hunk 1]
```diff
@@ -3,16 +3,18 @@ using DiffEqSensitivity,OrdinaryDiffEq, RecursiveArrayTools, DiffEqBase,
 using Test
 using DiffEqSensitivity: ForwardSensitivity
 
+using OrdinaryDiffEq
+using DiffEqSensitivity
```
Comment (ChrisRackauckas) [inline test/forward.jl:7]:
```suggestion
```

**Comment by ArnoStrouwen** [comment]
For the test error, lower tol or increase iterations?

**Comment by ChrisRackauckas** [comment]
It's off by 10^10. Something looks bad there.

**Comment by ArnoStrouwen** [comment]
Is it perhaps because the stiff solvers work with the same jacobian object?
https://github.com/SciML/SciMLBase.jl/blob/4edec99cdabcec8266a628dc49e778b4a3ce41a7/src/function_wrappers.jl#L15
And messing with this object inside the (extended) ODE system causes trouble?
Everything works fine on Tsit5(), so it is specifically the stiff solvers causing the issue.

**Comment by ArnoStrouwen** [comment]
I think the issue was just that Rodas4(autodiff=false) was only able to solve until 0.1s before maximum iterations was reached. 
The massive error in the test then was due to extrapolation to 5s?

**Comment by ChrisRackauckas** [comment]
The fact that it diverged is pretty iffy.

**Comment by ArnoStrouwen** [comment]
I can't find any sensitivity specific cause for the divergence.
It could be that the finite difference based jacobian (of the extended system) is iffy?

**Comment by ChrisRackauckas** [comment]
It shouldn't be that bad on this system. Check the full forwarddiff one to see how stable the new model is.

**Comment by ArnoStrouwen** [comment]
I tested if it was the finite difference based jacobian of the extended system by using ForwardDiffSensitivity instead of ForwardSensitivity:
```jl
probInpl3 = ODEForwardSensitivityProblem(fb,[1.0;1.0],(0.0,10.0),p,ForwardDiffSensitivity())
solInpl3 = solve(probInpl3,Rodas4(),abstol=1e-10,reltol=1e-10)
solInpl3 = solve(probInpl3,Rodas4(autodiff=false),abstol=1e-10,reltol=1e-10)      
```

```jl
julia> solInpl3 = solve(probInpl3,Rodas4(),abstol=1e-10,reltol=1e-10)
retcode: Success
Interpolation: specialized 3rd order "free" stiffness-aware interpolation
t: 4167-element Vector{Float64}:
  0.0
  0.0008227854154406597
  0.0018796649056780181
  0.0032608543701877967
  0.004916470576327273
  ⋮
  9.994272386460981
  9.996865914249916
  9.999470174260512
 10.0
u: 4167-element Vector{Vector{ForwardDiff.Dual{ForwardDiff.Tag{ODEFunction{true, typeof(fb), 
...

julia> solInpl3 = solve(probInpl3,Rodas4(autodiff=false),abstol=1e-10,reltol=1e-10)
retcode: Success
Interpolation: specialized 3rd order "free" stiffness-aware interpolation
t: 415556-element Vector{Float64}:
  0.0
  3.1827379411029245e-5
  4.3374170126018126e-5
  5.827269493778959e-5
  7.317121974956106e-5
  ⋮
  9.999858353275464
  9.999911268508027
  9.999964187597156
 10.0
u: 415556-element Vector{Vector{ForwardDiff.Dual{ForwardDiff.Tag{ODEFunction{true, typeof(fb), 
...
```

**Comment by ChrisRackauckas** [comment]
Try TRBDF2? That does look unstable under finite differences.

**Comment by ArnoStrouwen** [comment]
TRBDF2 works fine?
```jl
julia> solInpl3 = solve(probInpl3,TRBDF2(autodiff=false),abstol=1e-10,reltol=1e-10)
retcode: Success
Interpolation: 3rd order Hermite
t: 2959-element Vector{Float64}:

julia> solInpl3 = solve(probInpl3,TRBDF2(),abstol=1e-10,reltol=1e-10)
┌ Warning: `vendor()` is deprecated, use `BLAS.get_config()` and inspect the output instead
│   caller = isopenblas at linear_nonlinear.jl:86 [inlined]
└ @ Core C:\Users\arno\.julia\packages\DiffEqBase\uQlhE\src\linear_nonlinear.jl:86
retcode: Success
Interpolation: 3rd order Hermite
t: 2959-element Vector{Float64}:
```

**Comment by ChrisRackauckas** [comment]
Yup, that's expected. So yes, it is a Jacobian accuracy with finite difference issue (SDIRK methods do not need accurate Jacobians, while non-W Rosenbrock methods, such as Rodas4, do).

**Comment by ArnoStrouwen** [comment]
I rebased, please merge in if everything looks fine to you.

**Comment by ChrisRackauckas** [comment]
There's no change to the package code?

**Comment by ArnoStrouwen** [comment]
No the change to package code was already included in https://github.com/SciML/DiffEqSensitivity.jl/pull/470
Which is why I had to rebase, since I wrote this fix before that pr was completed.
Thus all that remains is writing a test.

**Pull request merged**